tests/test_mig_shared_configuration.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_install_geneateconfs - unit test of generateconfs
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests for the configuration object"""
+
+import inspect
+import io
+import json
+import unittest
+
+from tests.support import MigTestCase, PY2, testmain, fixturefile
+from mig.shared.configuration import Configuration
+
+
+def _is_method(value):
+    return type(value).__name__ == 'method'
+
+
+def _to_dict(obj):
+    return {k: v for k, v in inspect.getmembers(obj) if not (k.startswith('__') or _is_method(v))}
+
+
+class MigSharedConfiguration(MigTestCase):
+    @unittest.skipIf(PY2, "Python 3 only")
```

Comment:
Alright, I'll add the follow-up marker for this part.

Reply:
I don't remember if the intention was to follow-up or if I'd concluded that it was too difficult to make the _test_ work across both versions - may be an approach with completely fresh eyes and see what we conclude now type thing.